RecoLocalTracker/SiPixelRecHits/src/SiPixelTemplate.cc

```diff
@@ -213,6 +219,10 @@ bool SiPixelTemplate::pushfile(int filenum, std::vector< SiPixelTemplateStore >
       
       // next, layout the 1-d/2-d structures needed to store template
       
+      theCurrentTemp.cotbetaY = new float[theCurrentTemp.head.NTy];
+      theCurrentTemp.cotbetaX = new float[theCurrentTemp.head.NTyx];
+      theCurrentTemp.cotalphaX = new float[theCurrentTemp.head.NTxx];
```

Comment:
where are these deleted?

Reply:
please allocate on the stack  or at least use unique_ptr<float[]>

Reply:
This part of the code is for our [non-production] version of the template reco that is used for phase 2 studies and not ran in standard reconstruction (SI_PIXEL_TEMPLATE_USE_BOOST is not defined in CMSSW).

It can't be trivially changed to a unique_ptr, since that would require changing the declaration in the header from a float* to a unique_ptr or boost::multi_array and correspondingly the code in several places (e.g. SiPixelTemplate.cc#L2491) making it conflict with the default code in reconstruction.

Reply:
@ahinzmann can't you just delete them somewhere when not needed any more, still within some SI_PIXEL_TEMPLATE_USE_BOOST clause?

Reply:
yes, but still need to figure where, since it may be different in local running for phase 2 studies and standard reconstruction.

Reply:
These arrays are deleted in the destructor of PixelCPETemplateReco, which owns the vector of SiPixelTemplateStore.